src/views/Home/components/UserBanner/HarvestCard.tsx

```diff
@@ -24,13 +24,31 @@ const HarvestCard = () => {
   const masterChefContract = useMasterchef()
   const cakePriceBusd = usePriceCakeBusd()
   const earningsBusd = new BigNumber(farmEarningsSum).multipliedBy(cakePriceBusd)
-  const numFarmsToCollect = farmsWithStakedBalance.length
+  const numTotalToCollect = farmsWithStakedBalance.length
+  const numFarmsToCollect = farmsWithStakedBalance.filter((value) => value.pid !== 0).length
+  const hasCakePoolToCollect = numTotalToCollect - numFarmsToCollect > 0
 
-  const earningsText = t('%earningsBusd% to collect from %count% %farms%', {
+  let earningsText = ''
```

Comment:
This will show empty text, so just show `$0` with no context (while loading or when user has no stakes).
It could be changed to this:
```javascript
let earningsText = t('0 to collect') // needs to be added to translations.json
```
And then in the code below some change is needed too because `earningsBusd.toString()` might be `NaN` while loading. If user has real `0` `earningsBusd.toString()` will be `0` (just confirmed), so no need to check for that.
```javascript
const [preText, toCollectText] = earningsText.split(earningsBusd.isNaN() ? '0' : earningsBusd.toString())
```
and `preText` is not really shown so it doesn't matter, this `0` will never be displayed (see other comments about removing `preText`)

Reply:
Done